libtiff/libtiff_ctypes.py

```diff
@@ -823,6 +823,145 @@ def write_plane(arr, tile_arr, plane_index, image_width, image_height):
+            the returned dimensions will be (x, y, sample_index). 
+            If PlanarConfig == PLANARCONFIG_SEPARATE, 
+            the returned dimensions will be (sample_index, x, y).
+        """
+
+        num_tcols = self.GetField("TileWidth")
+        if num_tcols is None:
+            raise ValueError("TIFFTAG_TILEWIDTH must be set to read tiles")
+        num_trows = self.GetField("TileLength")
+        if num_trows is None:
+            num_trows = 1
+        num_irows = self.GetField("ImageLength")
+        if num_irows is None:
+            num_irows = 1
+        num_icols = self.GetField("ImageWidth")
+        if num_icols is None:
+            raise ValueError("TIFFTAG_IMAGEWIDTH must be set to read tiles")
+        # this number includes extra samples
+        samples_pp = self.GetField('SamplesPerPixel')
+        if samples_pp is None:  # default is 1
+            samples_pp = 1
+        planar_config = self.GetField('PlanarConfig')
+        if planar_config is None:  # default is contig
+            planar_config = PLANARCONFIG_CONTIG
+        num_idepth = self.GetField("ImageDepth")
+        if num_idepth is None:
+            num_idepth = 1
+        bits = self.GetField('BitsPerSample')
+        sample_format = self.GetField('SampleFormat')
+
+        # TODO: might need special support if bits < 8
+        dtype = self.get_numpy_type(bits, sample_format)
+
+        if y < 0 or y >= num_irows:
+            raise ValueError("Invalid y value")
+        if x < 0 or x >= num_icols:
+            raise ValueError("Invalid x value")
+
+        if y % num_trows != 0:
+            raise ValueError("y parameter not a multiple of tile height (%d)" % (num_trows))
+        if x % num_tcols != 0:
+            raise ValueError("x parameter not a multiple of tile width (%d)" % (num_tcols))
+
+        # if the tile is in the border, its size should be smaller
+        this_tile_height = min(num_trows, num_irows - y)
+        this_tile_width = min(num_tcols, num_icols - x)
+
+        def read_plane(tile_plane, plane_index):
+            """ Read one plane from TIFF. The TIFF has more than one plane only if
+            it has more than one sample per pixel, and planar_config == PLANARCONFIG_SEPARATE
+            """
+            # the numpy array should be contigous in memory before calling ReadTile
+            tile_plane = np.ascontiguousarray(tile_plane)
+            # even if the tile is on the edge, and the final size will be smaller,
+            # the size of the array passed to the ReadTile function
+            # must be (num_tcols, num_trows)
+            #
+            # The image has only one depth (ImageDepth == 1), so 
+            # the z parameter is always 0
+            r = self.ReadTile(tile_plane.ctypes.data, x, y, 0, plane_index)
+            if not r:
+                raise ValueError(
+                    "Could not read tile x:%d,y:%d,z:%d from file" % (x, y, z))
```

Comment:
No, just put in the error message, the same values as used in the call to ReadTile(). So I'd suggest to also indicate plane_index. So something like` "Could not read tile x:%d,y:%d,z:%d,s:% from file" %  (x,y,0,plane_index)`.